features/auth/LoginScreen.tsx

```diff
@@ -0,0 +1,192 @@
+import React, { useState } from 'react';
+import { Alert, StyleSheet, Text, View, KeyboardAvoidingView, ViewStyle } from 'react-native';
+import AppInput from '../../components/AppInput'
+import AuthButton from '../../components/AuthButton';
+import auth from '@react-native-firebase/auth';
+import { Screen } from '../../utils/navigation/Screens/Screens';
+import Colors from '../../utils/const/colors/Colors';
+import { ConstantLabels, ConstantText } from '../../utils/const/constantTexts/ConstantText';
+import getLogo from '../../assets/svgxml/logo';
+import { SvgXml } from 'react-native-svg';
+
+
+import { useNavigation } from '@react-navigation/native';
+import { AppStackParams } from '../../utils/navigation/AppNavigation';
+import { NativeStackNavigationProp } from '@react-navigation/native-stack';
+
+
+const LoginScreen = () => {
+
+    const [email, setEmail] = useState('');
+    const [password, setPassword] = useState('');
+
+    const [emailColor, setEmailColor] = useState(Colors.black);
+    const [passwordColor, setPasswordColor] = useState(Colors.black);
+
+    const [emailErrorMessage, setEmailErrorMessage] = useState('');
+    const [passwordErrorMessage, setPasswordErrorMessage] = useState('');
+
+    const [isHiddenMail, setIsHiddenMail] = useState(true);
+    const [isHiddenPass, setisHiddenPass] = useState(true);
+
+    const navigation = useNavigation<NativeStackNavigationProp<AppStackParams>>();
+
+    const handleLoginPressed = () => {
+        setIsHiddenMail(true);
+        setisHiddenPass(true);
+        setEmailColor(Colors.black);
```

Comment:
seems like whenever there is an error the email color is err_msg and if not than black. 
instead of having the color in a state. the error message is also an indicator, whenever its empty there is no error and whenever it's not there is an error. 
style={color: emailErrorMessage !== '' ? colors.err_msg : colors.black}
would work just as well and you can remove the state from the memory and having to set it every time

Reply:
same goes for password

Reply:
just make sure you set the errorMessage back to '' whenever there is not error